docs/source/en/model_doc/gpt2.md

```diff
@@ -60,6 +60,94 @@ different sizes: small, medium, large, xl and a distilled version of the small c
 
 This model was contributed by [thomwolf](https://huggingface.co/thomwolf). The original code can be found [here](https://openai.com/blog/better-language-models/).
 
+## Training
+
+GPT-2 is a decoder-only Transformer. It is trained using teacher forcing. This means that during training, we feed the ground truth
+previous tokens to the model in order to predict the next one. During training, one needs to provide the input sequences to the model as
+`input_ids`, as well as `labels`. The latter are identical to the `input_ids`. In teacher-forcing style, the model will internally shift the
+logits before calculating the loss. This ensures that the model is trained to predict the next token.
+
+One can use [`GPT2LMHeadModel`] (or the Tensorflow/Flax variant), which includes the language modeling head on top of the Transformer decoder.
+
+Suppose that we want to train the model for causal language modeling, then examples should be prepared for the model as follows:
+
+```python
+>>> from transformers import AutoTokenizer, AutoModelForCausalLM
+
+>>> tokenizer = AutoTokenizer.from_pretrained("gpt2")
+>>> model = AutoModelForCausalLM.from_pretrained("gpt2")
+
+>>> input_ids = tokenizer("Hugging Face is a company based in Paris and New York.", return_tensors="pt").input_ids
+>>> labels = input_ids.clone()
+
+>>> # the forward function internally shifts the logits to predict the next tokens and computes a cross-entropy loss
+>>> loss = model(input_ids=input_ids, labels=labels).loss
+>>> loss.item()
+4.075357437133789
+```
+
+As you can see, only 2 inputs are required for the model in order to compute a loss: `input_ids` (which are the
+`input_ids` of the encoded text) and `labels` (which are an exact copy of the `input_ids`, representing the text to be predicted).
+
+However, the example above only shows a single training example. In practice, one trains deep learning models in
+batches. What is typically done is one takes a large corpus of text, concatenates all text into one long sequence, and then chunks them up
+into smaller blocks. Hence GPT-2 style models typically don't require padding or truncation of text during training, as one simply trains on blocks of
+continuous text.
```

Comment:
```suggestion
However, the example above only shows a single training example. In practice, deep learning models are trained in
batches. Typically, you take a large corpus of text, concatenate all of it into one long sequence, and chunk it 
into smaller blocks. That's why GPT-2 style models typically don't require padding or truncating text during training because it trains on blocks of continuous text.
```